riscv-executor/src/lib.rs

```diff
@@ -645,59 +904,202 @@ struct Executor<'a, 'b, F: FieldElement> {
+            };
+        }
+        macro_rules! get_col {
+            ($name:ident) => {
+                self.proc.get_col(concat!("main::", stringify!($name)))
+            };
+        }
+
         let args = args
             .iter()
             .map(|expr| self.eval_expression(expr)[0])
             .collect::<Vec<_>>();
 
         self.proc.backup_reg_mem();
 
-        match name {
+        set_col!(X, get_col!(X_const));
+        set_col!(Y, get_col!(Y_const));
+        set_col!(Z, get_col!(Z_const));
+        set_col!(W, get_col!(W_const));
+        self.proc
+            .set_col(&format!("main::instr_{name}"), Elem::from_u32_as_fe(1));
+
+        let r = match name {
             "set_reg" => {
                 let addr = args[0].u();
-                self.proc.set_reg_mem(addr, args[1]);
+                let val = args[1];
+                self.reg_write(0, addr, val, 3);
+
+                set_col!(Y, val);
+
+                if !get_col!(Y_read_free).is_zero() {
+                    set_col!(Y_free_value, val);
+                }
 
                 Vec::new()
             }
             "get_reg" => {
                 let addr = args[0].u();
-                let val = self.proc.get_reg_mem(addr);
+                let val = self.reg_read(0, addr, 0);
 
                 vec![val]
             }
             "affine" => {
-                let val = self.proc.get_reg_mem(args[0].u());
+                let read_reg = args[0].u();
+                let val1 = self.reg_read(0, read_reg, 0);
                 let write_reg = args[1].u();
                 let factor = args[2];
                 let offset = args[3];
 
-                let val = val.mul(&factor).add(&offset);
+                let res = val1.mul(&factor).add(&offset);
 
-                self.proc.set_reg_mem(write_reg, val);
+                self.reg_write(1, write_reg, res, 3);
+                set_col!(tmp1_col, val1);
 
                 Vec::new()
             }
 
             "mstore" | "mstore_bootloader" => {
-                let addr1 = self.proc.get_reg_mem(args[0].u());
-                let addr2 = self.proc.get_reg_mem(args[1].u());
+                let read_reg1 = args[0].u();
+                let read_reg2 = args[1].u();
+                let addr1 = self.reg_read(0, read_reg1, 0);
+                let addr2 = self.reg_read(1, read_reg2, 1);
                 let offset = args[2].bin();
-                let value = self.proc.get_reg_mem(args[3].u());
+                let read_reg3 = args[3].u();
+                let value = self.reg_read(2, read_reg3, 2);
 
                 let addr = addr1.bin() - addr2.bin() + offset;
                 let addr = addr as u32;
                 assert_eq!(addr % 4, 0);
                 self.proc.set_mem(addr, value.u());
 
+                // Calculate the address with wrapping
+                let addr_full = addr1.bin() - addr2.bin() + offset;
+                // let addr_wrapped = (addr_full % (2_u64.pow(32) as i64)) as u32;
```

Comment:
leftover?

Reply:
mmm no, i think it should do the modulo to match the asm, ill uncomment